### DISCUSSION TITLE
15th open maintainers call

### DISCUSSION THREAD

**Discussion by mxinden** [libp2p/rust-libp2p#3529] [General]
Hi everyone,

The rust-libp2p maintainers (@mxinden and potentially others) will meet tomorrow (Tuesday). See [libp2p calendar](https://calendar.google.com/calendar/embed?src=libp2p.io_0q9682i3te7eanhe9q7ae1c58g%40group.calendar.google.com).

https://meet.jit.si/rust-libp2p-open-maintainers-call

Everyone is welcome to join, listen along, propose topics and join the discussions.

Please post below in case you already want to add an agenda item.

Items thus far:

- v0.51.0 release https://github.com/libp2p/rust-libp2p/releases/tag/v0.51.0
- libp2p perf implementation https://github.com/libp2p/rust-libp2p/pull/3508

See you all Tomorrow!

**Comment by thomaseizinger** [comment]
I am afraid I am not gonna be able to make this one. Some async input from my end:

- `NetworkBehaviour`s can now manage connections: https://github.com/libp2p/rust-libp2p/pull/3254. This is part of the `v0.51` release.
- An example for what this looks like can be seen in https://github.com/libp2p/rust-libp2p/pull/3386.
- Made progress on making our CI even faster: https://github.com/libp2p/rust-libp2p/pull/3469
- Some investigation on reducing our dependency tree. Sent a few PRs for the low-hanging fruit: https://github.com/libp2p/rust-libp2p/issues/3515

**Reply by mxinden** [reply]
A couple of pointers from yesterday's meeting:

- IPFS Thing https://discuss.ipfs.tech/t/pre-registration-is-open-for-ipfs-thing-april-2023/15707
- v0.51.0 release
- QUIC pull request for us to use quinn directly https://github.com/quinn-rs/quinn/pull/1219/
- https://github.com/webrtc-rs/webrtc/issues/413
- libp2p perf protocol implementation
    - https://github.com/mxinden/libp2p-perf/
    - https://github.com/libp2p/rust-libp2p/pull/3508
    - `docker run -ti --rm --entrypoint perf-client mxinden/libp2p-perf --server-address /dns4/libp2p-perf.max-inden.de/tcp/4001`
    - `docker run -ti --rm --entrypoint perf-client mxinden/libp2p-perf --server-address /dns4/libp2p-perf.max-inden.de/udp/4001/quic`
    - results from Nazar:
        ```nazar-pc@nazar-pc:~> docker run -ti --rm --entrypoint perf-client mxinden/libp2p-perf --server-address /dns4/libp2p-perf.max-inden.de/tcp/4001
        Finished run: Sent 10 MiB in 7.8186759 s with 10.231911518419635 MiBit/s and received 10 MiB in 7.96460177 s with 10.04444444433284 MiBit/s
        nazar-pc@nazar-pc:~> docker run -ti --rm --entrypoint perf-client mxinden/libp2p-perf --server-address /dns4/libp2p-perf.max-inden.de/udp/4001/quic
        Finished run: Sent 10 MiB in 13.62913457 s with 5.869778421301478 MiBit/s and received 10 MiB in 0.771775652 s with 103.6570663931751 MiBit/s```
    - results from Bogdan:
        - ```TCP: Finished run: Sent 10 MiB in 8.043945519 s with 9.945368204078212 MiBit/s and received 10 MiB in 8.248628169 s with 9.698582401938792 MiBit/s```
        - ```QUIC: Finished run: Sent 10 MiB in 5.521185882 s with 14.489640760115238 MiBit/s and received 10 MiB in 14.410963279 s with 5.551329113202162 MiBit/s```

- Learning resources
    - https://github.com/libp2p/rust-libp2p/tree/master/src/tutorials
    - Introduction to libp2p https://www.youtube.com/watch?v=Sbd7odDFT1w
    - https://docs.libp2p.io/concepts/introduction/overview/
